actionview/lib/action_view/helpers/url_helper.rb

```diff
@@ -743,7 +744,7 @@ def convert_options_to_data_attributes(options, html_options)
         end
 
         def url_target(name, options)
-          if name.respond_to?(:model_name) && options.empty?
+          if name.respond_to?(:model_name) && options.blank?
```

Comment:
Is this change just relying on the fact that `options` will always implement `blank?` (as opposed to `empty?`) regardless of whether it is a Hash or model instance?  If so, it might be more clear to check for what we're really interested in:

```suggestion
          if name.respond_to?(:model_name) && options.is_a?(Hash) && options.empty?
```

Or just:

```suggestion
          if name.respond_to?(:model_name) && options == {}
```

Reply:
Yeah, that's right. I suppose this is a better way to do it.

Reply:
I'm not sure we should even be changing the type of `options` actually... if it's blank, let it be blank.

Reply:
> I'm not sure we should even be changing the type of `options` actually

I'm not sure what you mean.  We wouldn't be changing the type of `options`; rather, we would be checking whether `options` can be safely ignored (i.e. it is an empty Hash).  Unless I am misunderstanding the code?

Reply:
I was referring to https://github.com/rails/rails/blob/39d22f9bf263f08e605c7da479553d6564d11502/actionview/lib/action_view/helpers/url_helper.rb#L211

There's no reason for this line (with my `blank?`) change. The only other place that `options` is referred to is used is https://github.com/rails/rails/blob/39d22f9bf263f08e605c7da479553d6564d11502/actionview/lib/action_view/helpers/url_helper.rb#L753 which already checks for its type (and it would be the same outcome if it were nil, or an empty hash).

So I think it is better to just remove the `||=` line.

Reply:
> There's no reason for this line (with my `blank?`) change. The only other place that `options` is referred to is used is ...

I feel like relying on this is a bit fragile.  `link_to_remote_options?` is already one step removed from `options ||= {}` via `convert_options_to_data_attributes`:

https://github.com/rails/rails/blob/39d22f9bf263f08e605c7da479553d6564d11502/actionview/lib/action_view/helpers/url_helper.rb#L211-L215

The code in either `convert_options_to_data_attributes` or `link_to_remote_options?` could change.  `options ||= {}` ensures that neither `convert_options_to_data_attributes` nor `link_to_remote_options?` nor any code they call (either now or in the future) needs to check for `nil`.

Reply:
Yeahh, I get that what you're saying. I'm not as worried because the test coverage is quite good.

To the original point, it feels wrong to do `options.is_a?(Hash) && options.empty?`, because if you pass anything except a hash for `options`, the `url_for(name)` code path will be called. And thus your `options` will be ignored silently.

It is better IMO to only call `url_for(name)` if there's *nothing* passed for `options` (hence `options.blank?)`, and in all other cases, delegate to `url_for(options)`. Which will then raise if `options` is not valid.

Reply:
> it feels wrong to do `options.is_a?(Hash) && options.empty?`, because if you pass anything except a hash for `options`, the `url_for(name)` code path will be called. And thus your `options` will be ignored silently.

It is the opposite, no?

Using `options.is_a?(Hash) && options.empty?`, only empty Hashes will be ignored.  Any other value for `options` will trigger the `url_for(options)` path, which can then raise on bad input.

Using `options.blank?`, `link_to @workshop, "\n"` would trigger the`url_for(name)` path, and ignore the bad `"\n"` input.

Reply:
Wow yeah I'm having a slow day. Yes, it is the opposite.